File: ios/FitFuel/Models/UserModel.swift
import Foundation

struct UserData: Codable {
    let id: Int
    let name: String
    let email: String
    let height: Int
    let weight: Int
    let gender: Gender
    let activityLevel: ActivityLevel
    let dateOfBirth: Date
    let dietaryPreferences: [String]
    let mealStats: MealStats?
    
    struct MealStats: Codable {
        let totalMeals: Int
        let compliantMeals: Int
        let averageCompliance: Double
    }
}

struct Meal: Codable, Identifiable {
    let id: Int
    let name: String
    let ingredients: [String]
    let nutritionalInfo: NutritionalInfo
    let preparationTime: Int
    let difficulty: String
    let mealType: String
    let scheduledTime: Date
    
    struct NutritionalInfo: Codable {
        let calories: Int
        let protein: Double
        let carbs: Double
        let fat: Double
    }
} 

File: android/app/src/main/java/com/fitfuel/models/UserModel.kt
package com.fitfuel.models

import java.util.Date

data class UserData(
    val id: Int,
    val name: String,
    val email: String,
    val height: Int,
    val weight: Int,
    val gender: Gender,
    val activityLevel: ActivityLevel,
    val dateOfBirth: Date,
    val dietaryPreferences: List<String>,
    val mealStats: MealStats?
)

data class MealStats(
    val totalMeals: Int,
    val compliantMeals: Int,
    val averageCompliance: Double
)

data class Meal(
    val id: Int,
    val name: String,
    val ingredients: List<String>,
    val nutritionalInfo: NutritionalInfo,
    val preparationTime: Int,
    val difficulty: String,
    val mealType: String,
    val scheduledTime: Date
)

data class NutritionalInfo(
    val calories: Int,
    val protein: Double,
    val carbs: Double,
    val fat: Double
) 